Library/Tests/ViewModels/MessageThreadCellViewModelTests.swift

```diff
@@ -50,7 +50,20 @@ final class MessageThreadCellViewModelTests: TestCase {
 
     withEnvironment(currentUser: thread.lastMessage.sender) {
       self.vm.inputs.configureWith(messageThread: thread)
+
       self.replyIndicatorHidden.assertValues([true, false])
     }
   }
+
+  func testUnreadIndicatorHidden() {
+    let thread = MessageThread.template
+
+    self.vm.inputs.configureWith(messageThread: thread)
+
+    self.vm.inputs.setSelected(false)
+    self.unreadIndicatorHidden.assertValues([false])
+
+    self.vm.inputs.setSelected(true)
+    self.unreadIndicatorHidden.assertValues([false, true])
```

Comment:
we could add a further test for the caching logic by simulating what happens when the cell scrolls off the page and back on. we do that by doing a `configureWith` with a different message thread, and then re-doing `configureWith` with this message thread. that requires having two message threads with different ids. wanna give that a shot?